Source/Magick.NET/Shared/Helpers/Throw.cs

```diff
@@ -40,8 +40,11 @@ public static void IfNullOrEmpty(string paramName, Stream value)
         {
             IfNull(paramName, value);
 
-            if (value.CanSeek && value.Position == value.Length)
+            if (value.Length == 0)
```

Comment:
Good idea to throw two different exceptions. You do need to check is `value.CanSeek` is true when you try to access `value.Length`.

Reply:
Oops thanks for catching that, should be fixed now